pkg/azuredx/adxauth/adxcredentials/builder.go

```diff
@@ -0,0 +1,171 @@
+package adxcredentials
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/grafana/azure-data-explorer-datasource/pkg/azuredx/adxauth/adxcredentials/internal/maputil"
+	"github.com/grafana/grafana-azure-sdk-go/azcredentials"
+	"github.com/grafana/grafana-azure-sdk-go/azsettings"
+)
+
+func FromDatasourceData(data map[string]interface{}, secureData map[string]string) (azcredentials.AzureCredentials, error) {
+	var credentials azcredentials.AzureCredentials
+	var err error
+
+	// Authentication types specific to the Azure Data Explorer datasource
+	if credentialsObj, err := maputil.GetMapOptional(data, "azureCredentials"); err != nil {
```

Comment:
Moved credentials definitions and parsing to the SDK and referenced them from there. ADX still has parsing of legacy configuration since it's ADX specific.